src/test/groovy/com/lesfurets/jenkins/TestRunClosure.groovy

```diff
@@ -0,0 +1,47 @@
+package com.lesfurets.jenkins
+
+import com.lesfurets.jenkins.unit.BasePipelineTest
+import org.junit.Before
+import org.junit.Test
+
+import static com.lesfurets.jenkins.unit.MethodCall.callArgsToString
+import static org.junit.Assert.assertTrue
+
+class TestRunClosure extends BasePipelineTest {
+
+    @Override
+    @Before
+    void setUp() throws Exception {
+        super.setUp()
+    }
+
+    @Test
+    void should_execute_without_errors() throws Exception {
+        runScript({ script ->
+            script.echo 'Test'
+        })
+        printCallStack()
+    }
+
+    @Test
+    void should_print_property_value() {
+        runScript({ script ->
+            script.println 'value'
+        })
+
+        def value = 'value'
+        assertTrue(helper.callStack.findAll { call ->
+            call.methodName == 'println'
+        }.any { call ->
+            callArgsToString(call).contains(value)
+        })
+    }
+
+    @Test
+    void should_use_registered_method() {
+        helper.registerAllowedMethod("customMethod", [Map.class], null)
+        runScript({ script ->
+            script.customMethod test: 'value'
+        })
+    }
```

Comment:
could you please add a test case with a shared library?

Reply:
I tried but I am not sure how to implement this. Could you take a look at my last commit?

Reply:
Just tried locally your changes, I was unable to load library with `@Library` annotation, but successfully loaded it dynamically.

Good news classes are working fine, but methods from `vars` not exists.

I believe you have to dig into groovy interceptors    

```
@Test
    void should_use_library_sayHello() {
        String sharedLibs = this.class.getResource('/libs/commons@master').getFile()

        def library = library('commons')
                .defaultVersion('<notNeeded>')
                .allowOverride(true)
                .implicit(false)
                .targetPath('<notNeeded>')
                .retriever(projectSource(sharedLibs))
                .build()

        helper.registerSharedLibrary(library)

        helper.registerAllowedMethod("library", [String.class], {String expression ->
            helper.getLibLoader().loadLibrary(expression)
            println helper.getLibLoader().libRecords
            return new LibClassLoader(helper,null)
        })

        runScript({ script ->
            def lib = script.library 'commons'
            def utils = lib.net.courtanet.jenkins.Utils.new(script)
            assert utils.gitTools() == [branch: 'master']
            script.sayHello()
        })
    }
```

Reply:
I can only suggest to look at https://github.com/jenkinsci/JenkinsPipelineUnit/blob/master/src/main/groovy/com/lesfurets/jenkins/unit/PipelineTestHelper.groovy#L355
it calls private instance of GroovyScriptEngine which contains all the groovy magic with pipeline methods interception and loading of shared libraries.

the workaround might be to save your closure to a `String` and load it via `loadScript`

Reply:
no need to save closure in a file. We just need an instance of Script (the script could be empty) loaded with JPU's custom `GroovyScriptEngine`

so mocked pipeline methods would work, and library classes would be available. And then we just rehydrate the closure to this empty script. (Inspired with previously implemented support 
 of declarative pipeline ) #13